File: apps/minecraft/spigot/src/main/java/crabcraft/net/crabUtilities/PackJoinListener.java
package crabcraft.net.crabUtilities;

import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PackJoinListener implements Listener {
    private final CrabUtilities plugin;

    public PackJoinListener(CrabUtilities plugin) {
        this.plugin = plugin;
    }

    @EventHandler(priority = EventPriority.MONITOR)
    public void onJoin(PlayerJoinEvent event) {
        plugin.getResourcePackManager().sendIfOptedIn(event.getPlayer());
    }
}

File: apps/minecraft/src/main/kotlin/crabcraft/net/crabUtilities/PackJoinListener.kt
package crabcraft.net.crabUtilities

import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

open class PackJoinListener(private val plugin: CrabUtilities) : Listener {
    @EventHandler(priority = EventPriority.MONITOR)
    open fun onJoin(event: PlayerJoinEvent) {
        plugin.getResourcePackManager().sendIfOptedIn(event.player)
    }
}
